### PR TITLE
Small update to Batch REST API specification, add some linter rule ignores to readme.md

### PR DESCRIPTION
**Pull request by matthchr**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by balajikris** [comment]
Hey @matthchr - so, i noticed the bug you linked to this PR and I need a small clarification. 

Did adding the "ignore" work  -- in the sense did they significantly impact the output of autorest linter before and after the change? Or did they make no impact because of the bug you linked to this PR? I'm asking because I still see a lot of linter errors and I am guessing those are the ones you are trying to suppress?

File: specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json [hunk 1]
```diff
@@ -12412,13 +12412,14 @@
         },
         "validationStatus": {
           "type": "string",
-          "title": "Whether the task counts have been validated. If the validationStatus is unvalidated, then the Batch service has not been able to check state counts against the task states as reported in the List Tasks API. The validationStatus may be unvalidated if the job contains more than 200,000 tasks.",
+          "title": "Whether the task counts have been validated. If the validationStatus is unvalidated, then the Batch service has not been able to check state counts against the task states as reported in the List Tasks API.",
+          "description": "The validationStatus may be unvalidated if the job contains more than 200,000 tasks.",
```
Comment (balajikris) [inline specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json]: _nit_: did you mean _invalidated_ in lines 12415 and 12416?

Comment (balajikris) [inline specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json]: english stackexchange tells me `unvalidated` does mean something😄 . I'd still like to hear from you what it means in this context.

**Comment by matthchr** [comment]
@balajikris For the ignore rules, the ones which I left in the file worked. I tried to ignore others, but they didn't work consistently (I tried for quite a while to get them to work too).

File: specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json (hunk 1 above)
Comment (matthchr) [inline specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json]: I updated the document (moved some text around) to help clarify.

`Validated` == Counts have been validated and confirmed to match the result of the List Tasks API.
`Unvalidated` == Counts have not been validated, and may not exactly match the result of the List Tasks API.

**Comment by matthchr** [comment]
Is there a reason I didn't get the Azure PR bot responding and saying how many warnings/errors I have?

File: specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json (hunk 1 above)
Comment (balajikris) [inline specification/batch/data-plane/Microsoft.Batch/2017-06-01.5.1/BatchService.json]: Thanks! I get it now 😄

**Review by balajikris — APPROVED** [review]

**Pull request merged**